be/test/storage/lake/compaction_task_test.cpp

```diff
@@ -364,9 +396,14 @@ TEST_F(DuplicateKeyOverlapSegmentsHorizontalCompactionTest, test) {
     ASSERT_TRUE(st.is_end_of_file());
 }
 
-class UniqueKeyHorizontalCompactionTest : public LakeCompactionTest {
+INSTANTIATE_TEST_SUITE_P(DuplicateKeyOverlapSegmentsCompactionTest, DuplicateKeyOverlapSegmentsCompactionTest,
+                         ::testing::Values(CompactionParam{HORIZONTAL_COMPACTION, 5},
```

Comment:
looks like `CompactionParam.algorithm` are not used in test case?

Reply:
used in `check_task` function

Reply:
ok. I figured it myself. the magic hides in `CompactionUtils::choose_compaction_algorithm`